core-rust/state-manager/src/jni/test_state_reader.rs

```diff
@@ -271,7 +271,7 @@ extern "system" fn Java_com_radixdlt_testutil_TestStateReader_leastStaleStateHas
     jni_sbor_coded_call(&env, request_payload, |_: ()| -> u64 {
         let database = JNINodeRustEnvironment::get_database(&env, j_rust_global_context);
         let least_stale_state_version = database
-            .access_direct() // the `get_stale_tree_parts_iter()` is inside a trait requiring writeability
+            .lock() // the `get_stale_tree_parts_iter()` is inside a trait requiring writeability
```

Comment:
_(note from author)_

The `StateHashTreeGcTest` that used this test-util became flakey (at least on my machine). I suspect it was caused by some unfortunate timings after I moved some code around (the test-util could not measure the exactly right version difference while commit kept pushing `state_version` forward).

Reply:
:+1: I also noticed it became flakey